x/claims/keeper/claim.go

```diff
@@ -136,3 +133,65 @@ func (k Keeper) ClaimCoinsForAction(ctx sdk.Context, addr sdk.AccAddress, action
 
 	return claimableAmount, nil
 }
+
+// MergeClaimsRecords merges two claim records from the
+// sender and recipient of the IBC transfer while claiming the
+// amount for the all the sender actions on behalf of the recipient.
+func (k Keeper) MergeClaimsRecords(
```

Comment:
this is the merge logic for when the user has claim records for the sender and recipient of the ibc transfer